### DISCUSSION TITLE
Order of keys in MultipartFormDataOutput

### DISCUSSION THREAD

**Discussion by GigiSan** [resteasy/resteasy#3253] [Q&A]
Hello there!

I'm trying to send a document to a remote API with RestEasy on Quarkus.

The API I'm calling requires me to send a multipart/form-data body with 3 files. Problem is, the API cares about the order in which they are specified and will return an error if they are in the wrong order.

The parameters are called PARAMETERS, INDEX and DATA and need to be specified in this exact order and intentionally uppercased. I'm using a [MultipartFormDataOutput](https://docs.jboss.org/resteasy/docs/3.0-beta-4/javadocs/index.html?org/jboss/resteasy/plugins/providers/multipart/MultipartFormDataOutput.html) to store the files to send, but the keys are sent in the wrong order as seem to get ordered alphabetically. Is there any way to alter the order? I tried reversing them, thinking maybe they got stacked, but the order is unchanged at call time.

```
MultipartFormDataOutput mpFormData = new MultipartFormDataOutput();
mpFormData.addFormData("PARAMETERS", requestData.getParamFile(), new MediaType("application", "xml"), "parametersFile.xml");
mpFormData.addFormData("INDEX", requestData.getIndexFile(), new MediaType("application", "xml"), "indexFile.xml");
mpFormData.addFormData("DATA", requestData.getDataFile(), new MediaType("application", "json"), "dataFile.json");
docuService.sendDocument(
        mpFormData
);
```

I also posted a [question on SO](https://stackoverflow.com/questions/73686447/order-of-keys-in-resteasy-multipartformdataoutput), hoping someone catches it and comes up with a suggestion :)

**Comment by GigiSan** [comment]
I've been given a suggestion on SO, so I'll try that, although it would be nice to have  away out of the box.
I can't see an option to close or delete the question, feel free to if you can :)

**Comment by jamezp** [comment]
This is a valid issue IMO. I filed https://issues.redhat.com/browse/RESTEASY-3217 to fix this. There is a change I may break the workaround I suggested on SO though as I think the map should be 1 private and there should only be 1 map, not two.
